### PR TITLE
Change the behaviour of find_input in list context

### PR DESCRIPTION
**Pull request by choroba**
and when given non-positive index: warns in the former case, dies in
the latter.

### CODE REVIEW

**Comment by choroba** [comment]
Fixes #7 
Greetings from the Pull Request Club.

File: lib/HTML/Form.pm [hunk 1]
```diff
@@ -506,7 +506,11 @@ input with the given name and/or type.
 sub find_input
 {
     my($self, $name, $type, $no) = @_;
+    die "Invalid index $no.\n"
```
Comment (simbabque) [inline lib/HTML/Form.pm]: All the existing calls to `die` do not have a newline at the end, so will include our line number. Is there a reason you've chosen to change that behaviour here?

Comment (choroba) [reply lib/HTML/Form.pm]: It was an oversight. I'll create a fixup.

**Comment by karenetheridge** [comment]
> `2976fc1 fixup! Change the behaviour of find_input in list context`

This looks like this should have been rebased into the previous commit as a fixup? (`git rebase -i master`; edit the last `pick` to 'fixup`, save file)

**Comment by choroba** [comment]
No. I rebased the original commit to upstream/master that changed in the meantime, and created a fixup that I'll squash once reviewers say the changes are OK. At least, that's how we use git and GitHub at work, sorry for any confusion.

**Comment by choroba** [comment]
Squashed, ready to merge.

**Comment by oalders** [comment]
Thanks @choroba, @simbabque and @karenetheridge!

**Pull request merged**